a10_octavia/controller/worker/tasks/nat_pool_tasks.py

```diff
@@ -0,0 +1,116 @@
+#    Copyright 2020, A10 Networks
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from oslo_log import log as logging
+from taskflow import task
+
+import acos_client.errors as acos_errors
+
+from a10_octavia.controller.worker.tasks.decorators import axapi_client_decorator
+
+LOG = logging.getLogger(__name__)
+
+
+class NatPoolCreate(task.Task):
+    """Task to create nat pool"""
+
+    @axapi_client_decorator
+    def execute(self, loadbalancer, vthunder, flavor_data=None):
+        device_pool = None
+        if flavor_data:
+            natpool_flavor_list = flavor_data.get('nat_pool_list')
+            natpool_flavor = flavor_data.get('nat_pool')
+            if natpool_flavor_list:
+                for i in range(len(natpool_flavor_list)):
+                    device_pool = self.axapi_client.nat.pool.try_get(
+                        natpool_flavor_list[i]['pool_name'])
+                    if not (device_pool and
+                            (device_pool['pool']['start-address'] ==
+                                natpool_flavor_list[i]['start_address'] and
+                                device_pool['pool']['end-address'] ==
+                                natpool_flavor_list[i]['end_address'])):
+                        pool_name = natpool_flavor_list[i]['pool_name']
```

Comment:
We don't need to do all of this variable creation. The flavor only has this information. Just use:

```
                            self.axapi_client.nat.pool.create(**natpool_flavor_list[i])
```

Reply:
If more info is added, then it will be required in this nat pool create step anyway

Reply:
Instead of `self.axapi_client.nat.pool.create(**natpool_flavor_list[i])`, adding the below call as create() requires 5 arguments to be passed:
`self.axapi_client.nat.pool.create(pool_name, start_address, end_address, netmask, **natpool_flavor_list[i])` 

Please refer the create function definition on acos side https://github.com/a10networks/acos-client/blob/feature/snat/acos_client/v30/nat.py#L71

Reply:
And removing the variable created for `gateway`

Reply:
We should definitely match the function params to the payloads. That's what is done pretty everywhere else so that it's possible to do these splat operations.

Reply:
yes changed the call to `self.axapi_client.nat.pool.create(**natpool_flavor_list[i])` after modifying the payload on acos side